src/Compilers/Core/Portable/MetadataReference/ModuleMetadata.cs

```diff
@@ -164,6 +174,16 @@ public static ModuleMetadata CreateFromStream(Stream peStream, PEStreamOptions o
                 throw new ArgumentNullException(nameof(peStream));
             }
 
+            // If this stream is an UnmanagedMemoryStream, we can heavily optimize creating the metadata by directly
+            // accessing the underlying memory.
+            if (peStream is UnmanagedMemoryStream unmanagedMemoryStream)
```

Comment:
>peStream [](http://example.com/codeflow?start=16&length=8)

`if (options & (PEStreamOptions.PrefetchEntireImage | `PEStreamOptions.PrefetchMetadata) == 0 && peStream is ...)`